lemmy_api/src/community.rs

```diff
@@ -129,15 +130,31 @@ impl Perform for CreateCommunity {
       return Err(APIError::err("community_already_exists").into());
     }
 
+    // Check to make sure the icon and banners are urls
+    let icon = Some(data.icon.to_owned());
+    let banner = Some(data.banner.to_owned());
+
+    if let Some(Some(icon)) = &icon {
+      if Url::parse(icon).is_err() {
+        return Err(APIError::err("invalid_url").into());
+      }
+    }
+
+    if let Some(Some(banner)) = &banner {
+      if Url::parse(banner).is_err() {
```

Comment:
You might consider taking the parsed value and assigning that to `banner`. That would give some normalisation, eg it removes the useless `:`. But its probably too complicated to be worth the effort.

Reply:
Hrm, looks like there is an `.as_str()` on the Url crate, but I don't know that it'd be a good idea to use it, it *could* manipulate it in unwanted ways. Something to keep in mind, but just an ParseError check should be fine for now.